### ISSUE TITLE
When using Misa Texture Pack it consumes all the RAM on the computer until it crashes

### ISSUE THREAD

**Issue by juliocoliveira** [Fabulously-Optimized/fabulously-optimized#554]
### Describe the bug

It runs fine on vanilla texture pack, If i add the texture pack that i use "Misa 64x" https://www.patreon.com/file?h=75386242&i=12445894 it consumes all the ram on the computer, I allocated 4GB of RAM, and my computer have 32GB.
That does happen on the 1.19.2 and 1.19 too, that does not happen on normal fabric instance or vanilla, Only on fabulously Optimized ones.
Tested on 1.18.2 and it works fine.

Here is the log:
```
https://paste.ee/p/NOvL3
```

Using the Modrinth version on MultiMC.

### Modpack version

4.6.0 alpha 12

### Launcher

MultiMC

### Other things

- [X] I updated FO to a newer version (mention, what version to what version)
- [ ] I added/removed some mods or resource packs (mention, which ones)

**Comment by Julienraptor01** [comment]
since it's a high res RP try with 6GB of ram instead
\+ this RP does seem to require very strict OF features 
and in addition, this pack uses broken path which is a very bad thing
i think it can't even work on 1.19.3 FO due to missing required features

**Comment by juliocoliveira** [comment]
I set 6GB on Allocation, but it does NOT matter, It still consumes ALL my RAM. I mean every single megabyte until it crashes, for some reason the memory allocation/limit is not being respected if I enable this resource pack on this mod pack.

**Comment by Julienraptor01** [comment]
as i said due to missing features in FO 1.19.3 (which is alpha) this RP can't work currently

**Comment by Kichura** [comment]
I don't think this is quite what is meant to happen; What the OP is referring is that FO has a possibility of causing actual memory leaks when using the resource pack, and yet there is no concrete evidence of what causes that to happen out of thin air.

**Comment by Julienraptor01** [comment]
> I don't think this is quite what is meant to happen; What the OP is referring is that FO has a possibility of causing actual memory leaks when using the resource pack, and yet there is no concrete evidence of what causes that to happen out of thin air.

what i'm saying is that what happen is expected since every required feature to correctly load this pack isn't present and by so this modpack has problems loading which causes OoM

**Comment by Kichura** [comment]
Well if it truly that then it would be resource pack creator's issue and not this, But if it was FO's alpha build then the only question i can throw out is: Does 4.5.5 fix anything?

**Comment by Julienraptor01** [comment]
> Well if it is truly that then it would be resource pack creator's issue and not this, But if it was FO's alpha build then the only question i can throw out is: Does 4.5.5 fix anything?

he said 1.18.2 work fine

**Comment by Kichura** [comment]
> > Well if it is truly that then it would be resource pack creator's issue and not this, But if it was FO's alpha build then the only question i can throw out is: Does 4.5.5 fix anything?
> 
> he said 1.18.2 work fine

Not even 4.5.5 works?!, Now i am actually curious to see where the problem really lays at.

**Comment by Julienraptor01** [comment]
> > > Well if it is truly that then it would be resource pack creator's issue and not this, But if it was FO's alpha build then the only question i can throw out is: Does 4.5.5 fix anything?
> > 
> > 
> > he said 1.18.2 work fine
> 
> Not even 4.5.5 works?!, Now i am actually curious to see where the problem really lays at.

>>> It runs fine on vanilla texture pack, If i add the texture pack that i use "Misa 64x" https://www.patreon.com/file?h=75386242&i=12445894 it consumes all the ram on the computer, I allocated 4GB of RAM, and my computer have 32GB.
That does happen on the 1.19.2 and 1.19 too, that does not happen on normal fabric instance or vanilla, Only on fabulously Optimized ones.
### **_Tested on 1.18.2 and it works fine._**

**Comment by Kichura** [comment]
> > > > Well if it is truly that then it would be resource pack creator's issue and not this, But if it was FO's alpha build then the only question i can throw out is: Does 4.5.5 fix anything?
> > > 
> > > 
> > > he said 1.18.2 work fine
> > 
> > 
> > Not even 4.5.5 works?!, Now i am actually curious to see where the problem really lays at.
> 
> > It runs fine on vanilla texture pack, If i add the texture pack that i use "Misa 64x" [patreon.com/file?h=75386242&i=12445894](https://www.patreon.com/file?h=75386242&i=12445894) it consumes all the ram on the computer, I allocated 4GB of RAM, and my computer have 32GB.
> > That does happen on the 1.19.2 and 1.19 too, that does not happen on normal fabric instance or vanilla, Only on fabulously Optimized ones.
> 
> ### **_Tested on 1.18.2 and it works fine._**

That is not what i am referring to, I was asking about 4.5.5 which is 1.19.2 itself; Not the alpha and not 1.18.2, Please re-read my comment one more time just to be sure.

**Comment by Julienraptor01** [comment]
wait sry i'm f blind

**Comment by Julienraptor01** [comment]
work fine on FO 1.19.2
i just tested

**Comment by Kichura** [comment]
Then from what i can tell; Either the OP's java arguments are doing weird shenanigans or the alpha build has issues working properly with high-resolution packs at the moment.

**Comment by juliocoliveira** [comment]
To me it does not work on 1.19.2: https://paste.ee/p/f8kkH

**Comment by Julienraptor01** [comment]
ok wait
it seems to crash half of the time on pure FO
i got it running once but now it crash

**Comment by Julienraptor01** [comment]
it seems like with my modified FO i can get it running consistently tho

**Comment by Kichura** [comment]
Can you provide the mods you have used to get it to run?

**Comment by Julienraptor01** [comment]
it uses 10GB on 4 allocated but at least it work XD

**Comment by Julienraptor01** [comment]
wait i have a F idea it may be due to the java arg i'm using to force the game to not crash when i encounter an error with the nvidia drivers

**Comment by Julienraptor01** [comment]
ok yes

**Comment by Kichura** [comment]
Cannot reproduce ([specs here](https://pastebin.com/raw/jMTU4fSW)):

![fo](https://user-images.githubusercontent.com/68134602/209479385-359e1287-c2f8-42eb-b9a2-4ad78ffa5798.png)

**Comment by Julienraptor01** [comment]
use the arg
`-XX:-CriticalJNINatives`
in your java args to force the game to not crash

**Comment by Julienraptor01** [comment]
> Cannot reproduce ([specs here](https://pastebin.com/raw/jMTU4fSW)):
> 
> ![fo](https://user-images.githubusercontent.com/68134602/209479385-359e1287-c2f8-42eb-b9a2-4ad78ffa5798.png)

i can

**Comment by Kichura** [comment]
I have used JDK 19.0.1 with the shenandoahgc set and it works fine, guessing G1GC / default java arguments bombs it?

**Comment by Julienraptor01** [comment]
+ if i clean the memory with the system management app from my pc, the ram is able to stay within limits
it just go to 10GB when loading

**Comment by Julienraptor01** [comment]
take note that the java arg i provided don't work with the vanilla launcher

**Comment by Kichura** [comment]
That is because the vanilla launcher is known for bricking newer java installations as the javacheck.jar program is badly programmed by Mojang themselves, This can be checked upon in [MCL-18306](https://bugs.mojang.com/browse/MCL-18306).

**Comment by Julienraptor01** [comment]
So, to recap, to use this pack you must use this java arg `-XX:-CriticalJNINatives` and expect the game to go to 10GB when loading
if you have an app like this by example, you can use it to clear the memory and it'll go down to a normal amount and stay like that
![image](https://user-images.githubusercontent.com/49841713/209479563-020e7915-35d9-439c-8092-58882512f98e.png)

**Comment by Kichura** [comment]
I see this as in being a morely hacky method than anything virtually fixable in practice and should be furtherly checked upon - Worst case scenario would be that windows in general has terrible memory management in first place if valid.

**Comment by Julienraptor01** [comment]
> I see this as in being a morely hacky method than anything virtually fixable in practice and should be furtherly checked upon - Worst case scenario would be that windows in general has terrible memory management in first place if valid.

this is an hacky method yes but the RP seems to have a LOT of issues of missing parts and also has very bad things like wrong number of tiles for CTM, bad naming scheme (broken paths), etc

**Comment by Julienraptor01** [comment]
this issue come from the pack creator not doing things properly which end up causing issues with FO
i tested with 1.19.3 and this method work too
<details>
  <summary>little info of why i know this ARG</summary>
(just so you know, i use and know this arg because NVIDIA drivers recents updates are so bad that they introduced a lot of crash bugs and BSoD that i have to force the game to not crahs)
  
</details>

**Comment by juliocoliveira** [comment]
What i can confirm for sure that the issue is with Fabulously Optimized +this texture pack "Misa 64x 1.19". I tested Clarity, FaithfulHDx512 and Epic Adventures, all worked fine, Maybe @Julienraptor01 is right and there is a lot of wrong things on this texture pack, I don't know, I don't understand how a texture pack is made.

I don't use any custom java args, only the defaults from MultiMC, and even if you managed to it to work, Using 10GB when you say to use only 4GB confirms the memory leak.

Maybe the issue is the size of the texture pack, Even the 512x is 27MB while Misa is 161MB. Misa is very rich, have a lot of Custom Mob Models. 

The texture pack version of the 1.18.2 is a different file than 1.19+.
https://mega.nz/file/pt8hgSaK#XWH3dKAnk62BpErY2M3bGyo2ojMtnWk_9lXYph-fKe4

Maybe there is something new on the texture pack or maybe a new mod or an updated mod is the issue.
I don't know, But the texture pack should work here, It works on 1.18.2 and Vanilla, Optifine or Standard Fabric (Non Mod Pack).

If you guys managed to fix it, would be great, if not, well, thanks anyway.

**Comment by Julienraptor01** [comment]
> What i can confirm for sure that the issue is with Fabulously Optimized +this texture pack "Misa 64x 1.19". I tested Clarity, FaithfulHDx512 and Epic Adventures, all worked fine, Maybe @Julienraptor01 is right and there is a lot of wrong things on this texture pack, I don't know, I don't understand how a texture pack is made.
> 
> I don't use any custom java args, only the defaults from MultiMC, and even if you managed to it to work, Using 10GB when you say to use only 4GB confirms the memory leak.
> 
> Maybe the issue is the size of the texture pack, Even the 512x is 27MB while Misa is 161MB. Misa is very rich, have a lot of Custom Mob Models.
> 
> The texture pack version of the 1.18.2 is a different file than 1.19+. https://mega.nz/file/pt8hgSaK#XWH3dKAnk62BpErY2M3bGyo2ojMtnWk_9lXYph-fKe4
> 
> Maybe there is something new on the texture pack or maybe a new mod or an updated mod is the issue. I don't know, But the texture pack should work here, It works on 1.18.2 and Vanilla, Optifine or Standard Fabric (Non Mod Pack).

what do you mean standard fabric ? fabric but whit no mods loaded ?
(it surely work on OF because OF has support for a lot of broken thing and it work on vanilla because vanilla doesn't try to load OF features
it's broken here because FO has all the mods to support proper non broken OF features and this RP use broken features that OF support)
TL;DR says to the creator to fix the RP and if he does that it should work properly too

> 
> If you guys managed to fix it, would be great, if not, well, thanks anyway.

the "leak" only come from the RP being broken i think
it isnt really a leak and more like this RP does need 10GB of RAM, aka if you put 12GB of RAM, the game will use 10 and not be OoM
you can add the java arg i provided here like this
![image](https://user-images.githubusercontent.com/49841713/209480818-7aceb8f2-2d83-4084-9695-b3ced2e05f8c.png)
to temporarily fix the issue but it would be better if the pack was fixed by its creator instead
as i said if you then use a tool to clear the system memorey (mine is provided by my pc manufacturer) you can clear the memory once the game is loaded and it'll stay at a normal amount

**Comment by Madis0** [comment]
Lack of response, so I presume this is fixed.

**Comment by juliocoliveira** [comment]
Nope, not fixed. Didn't said anything because there was nothing i could contribute with. The response above is very hacky workaround not a definite, good fix.

**Comment by Julienraptor01** [comment]
> Nope, not fixed. Didn't said anything because there was nothing i could contribute with. The response above is very hacky workaround not a definite, good fix.

did you report to the RP creator that the pack has many problems in itself ? because i think it's actually the biggest source of problems i think

**Comment by juliocoliveira** [comment]
No i didn't, What is EXACTLY the problems with this pack? What is broken paths?

**Comment by Julienraptor01** [comment]
> No i didn't, What is EXACTLY the problems with this pack? What is broken paths?

Broken path is when you use characters that aren't allowed to exist.
The game don't support them in vanilla, but for some reason OptiFine decided it would be a good idea to add support for it ! Except that it tend to break and that some OF parity mods have a very hard time to support it so it cause even more problems in fabric

> Legal characters :
> Namespaces, paths and other folder and file names in the resource/data pack should only contain the following symbols:
> 
>     0123456789 Numbers
>     abcdefghijklmnopqrstuvwxyz Lowercase letters
>     _ Underscore
>     - Hyphen/minus
>     / Forward Slash/Directory separator (Can't be used in namespace)
>     . Period
> 
> The preferred naming convention is lower_case_with_underscores, called lower snake case. 

and i also noted other problems that i mentioned here

> https://github.com/Fabulously-Optimized/fabulously-optimized/issues/554#issuecomment-1364726772
> this is an hacky method yes but the RP seems to have a LOT of issues of missing parts and also has very bad things like wrong number of tiles for CTM, bad naming scheme (broken paths), etc

from what i said up there the RP sometimes is missing some files or have a wrong number of files mentioned in properties files which surely induce even more problems
